pkgs/by-name/li/librandombytes/package.nix

```diff
@@ -0,0 +1,81 @@
+{
+  stdenv,
+  lib,
+  python3,
+  openssl,
+  fetchzip,
+}:
+stdenv.mkDerivation (prev: {
+  pname = "librandombytes";
+  version = "20240318";
+
+  src = fetchzip {
+    url = "https://randombytes.cr.yp.to/librandombytes-${prev.version}.tar.gz";
+    hash = "sha256-LE8iWw7FxckPREyqefgKtslD6CPDsL7VsfHScQ6JmLs=";
+  };
+
+  postPatch = ''
+    patchShebangs configure
+    patchShebangs scripts-build
+  '';
+
+  # NOTE: librandombytes uses a custom Python `./configure`: it does not expect standard
+  # autoconfig --build --host etc. arguments: disable
+  configurePlatforms = [ ];
+
+  # NOTE: the librandombytes library has required specific CFLAGS defined:
+  # https://randombytes.cr.yp.to/librandombytes-20240318/compilers/default.html
+  # - `-O` (alias `-O1`) safe optimization
+  # - `-Qunused-arguments` suppress clang warning
+  # the default "fortify" hardening sets -O2, -D_FORTIFY_SOURCE=2:
+  # since librandombytes uses -O1, we disable the fortify hardening, and then manually re-enable -D_FORTIFY_SOURCE.
+  hardeningDisable = [ "fortify" ];
+  env.NIX_CFLAGS_COMPILE = toString (
+    lib.optionals stdenv.cc.isClang [ "-Qunused-arguments" ]
+    ++ [
+      "-D_FORTIFY_SOURCE=2"
+      "-O1"
+    ]
+  );
+
+  patches = [ ./environment-variable-tools.patch ];
```

Comment:
I would put it just before `postPatch` so that the expression is in the execution order (unpack -> patch -> configure -> build), but it's optional anyway. LGTM otherwise.